FHEM/00_SIGNALduino.pm

```diff
@@ -295,7 +295,7 @@ sub SIGNALduino_Initialize {
 					  .' doubleMsgCheck_IDs'
 					  .' eventlogging:0,1'
 					  .' flashCommand'
-					  .' hardware:ESP8266,ESP8266cc1101,ESP32,nano328,nanoCC1101,miniculCC1101,promini,radinoCC1101'
+					  .' hardware:ESP32,ESP32cc1101,ESP8266,ESP8266cc1101,MAPLEMINI_F103CB,MAPLEMINI_F103CBcc1101,nano328,nanoCC1101,miniculCC1101,promini,radinoCC1101'
```

Comment:
Da sollten ein paar Tests erweitert werden und mit dem Hardware Attribut das Verhalten testen.

Zum einen im Bereich Flash könnte es interessante Ergebnisse geben aber ggf. auch dadurch dass das cc1101 mit kleinem cc geschrieben ist, könnte das ggf. an der ein oder anderen Stelle eine Auswirkung haben.
Ich bin mir da  zumindest nicht ganz sicher.

Reply:
Dann passe ich den Namen an um ein großes CC1101 zu erhalten.

Was du bei dem Namen testen möchtest, weiß ich nicht. Den Zusammenhang kenne ich nicht und nur deine Gedanken wissen dies.

Reply:
Beim Flashen wird das Hardware Attribut ausgewertet um das passende Vorgehen zu verwenden.

Ob da noch mehr stellen waren, da bin ich mir nicht zu 100% sicher.
Vermutlich erkennen die es alle aus dem Versionsstring :)

Reply:
Wenn ich das richtig überfliege, so werden hier https://api.github.com/repos/RFD-FHEM/SIGNALDuino/releases die Namen gelesen und vie Regex vom Attributnamen gefiltert. Da wäre die gleiche Schreibweise klein ;-)

Wie ist deiner Meinung nach der richtige Weg? Wenn wir überlegen, es ist ne kleine Änderung wo wir große Test aufbauen, so werden die Schritte ja klein vollzogen und die PR´s stauen sich *grins*

Reply:
Wenn wir den Test nicht jetzt erweitern, werden wir es nie machen.  Und wenn die Änderungen klein sind, dann staut sich das auch nicht an, weil nur wenig zu reviewen ist

Reply:
Okay, würdest du den Test in angriff nehmen? Das ist ja das "kleinste" Übel der PR.
Somit kommen wir Step for Step voran.

Reply:
Tests habe ich erweitert, mit dem Ergebnis was ich erwarten würde.

Die Tests schlagen allerdings fehl

1) Download der Firmware klappt nicht
2) Rückmeldung beim Ausführen des Flash Befehles passt nicht für den STM32.

Reply:
Kannst du verifizieren, das es nur der Attributname ist oder einfach versuchen und den Namen ändern?
WEnn dies angepasst werden muss, dann muss ich ja handeln.

Reply:
Für den einen Test muss ich das Json noch aktualisieren, dann läuft der durch.
Ist quasi schon erledigt.

Bei dem anderen Text passt der Rückgabetext nicht wenn der Flash Befehl auf den Maple ausgeführt wird.

Hier steht fest ESP und wir sollten das vielleicht durch den Wert ersetzen der im Attribut Hardware steckt.
Dann sieht es auch schon gut aus mit den Tests.

Reply:
Meinst du diese Stelle mit fest verankert?
https://github.com/RFD-FHEM/RFFHEM/blob/f3a4ff24df5f43e7fbb0aa2eefb97256d63e6691/FHEM/00_SIGNALduino.pm#L701-L702

Reply:
Ja das sollte die Stelle sein.